src/redirect.rs

```diff
@@ -0,0 +1,149 @@
+use anyhow::{anyhow, Result};
+use reqwest::blocking::Client;
+use reqwest::blocking::{Request, Response};
+use reqwest::header::{
+    HeaderMap, AUTHORIZATION, CONTENT_ENCODING, CONTENT_LENGTH, CONTENT_TYPE, COOKIE, LOCATION,
+    PROXY_AUTHORIZATION, TRANSFER_ENCODING, WWW_AUTHENTICATE,
+};
+use reqwest::{Method, StatusCode, Url};
+
+#[derive(Debug)]
+enum ClonedRequest {
+    Partial(Request),
+    Full(Request),
+}
+
+impl ClonedRequest {
+    fn inner(self) -> Request {
+        match self {
+            Self::Partial(request) | Self::Full(request) => request,
+        }
+    }
+}
+
+fn clone_request(request: &Request) -> ClonedRequest {
+    let mut cloned_request = Request::new(request.method().clone(), request.url().clone());
+    *cloned_request.timeout_mut() = request.timeout().cloned();
+    *cloned_request.headers_mut() = request.headers().clone();
+
+    match request.body().map(|b| b.as_bytes()) {
+        Some(Some(body)) => {
```

Comment:
I think the code in this file could be drastically simplified if we chose to buffer the body here which is an approach that something that some crates take e.g https://docs.rs/follow-redirects/0.1.3/follow_redirects/

Reply:
One problem I see with the current approach is that it's a bit unintuitive. For example, this following will successfully reach the last destination because the request body has been buffered by the `-v` flag
```
cargo run  -- --follow --all -v post pie.dev/redirect-to url==https://pie.dev/post status_code==307 @Cargo.toml
```

But this will stop midway
```
cargo run  -- --follow --all post pie.dev/redirect-to url==https://pie.dev/post status_code==307 @Cargo.toml
```

Reply:
We are now buffering the request body before following HTTP redirects.